projects/graph/graph.py

```diff
@@ -63,23 +63,56 @@ def dft_recursive(self, starting_vertex):
         beginning from starting_vertex.
         This should be done using recursion.
         """
-        pass  # TODO
+        def go(starting_vertex, acc):
+            if starting_vertex in acc:
+                return
+            print(starting_vertex)
+            for neighbor in self.vertices[starting_vertex]:
+                acc.add(starting_vertex)
+                go(neighbor, acc)
+        go(starting_vertex, set())
 
     def bfs(self, starting_vertex, destination_vertex):
         """
         Return a list containing the shortest path from
         starting_vertex to destination_vertex in
         breath-first order.
         """
-        pass  # TODO
+        visited = set()
+        q = Queue()
+        q.enqueue([starting_vertex])
+
+        while q.size() > 0:
+            vs = q.dequeue()
+            v = vs[-1]
+            if v not in visited:
+                visited.add(v)
+                for neighbor in self.vertices[v]:
+                    next_vs = vs + [neighbor]
+                    if neighbor == destination_vertex:
+                        return next_vs
+                    q.enqueue(next_vs)
 
```

Comment:
looks good; should add a return False outside the while loop in case the value cannot be found